packages/dapp/src/hooks/useKey.ts

```diff
@@ -0,0 +1,55 @@
+import type { KeyRecord } from '../store/actions';
+import type { JsonRpcSigner } from '@ethersproject/providers';
+import { useState, useMemo, useEffect } from 'react';
+import { findKeyById, findKeyByTag } from './useKeysManager';
+import { useAppState } from '../store';
+import Logger from '../utils/logger';
+
+// Initialize logger
+const logger = Logger('useKeysManager');
+const UNKNOWN_ERROR = 'Unknown useKeysManager error';
+
+export type UseKeyHook = [
+  key: KeyRecord | undefined,
+  signer: JsonRpcSigner | undefined,
+  error: string | undefined,
+];
+
+// UseKey react hook
+export const useKeysManager = (
+  keyIdOrTag: string
+): UseKeyHook => {
+  const { keys, provider } = useAppState();
+
+  const [key, setKey] = useState<KeyRecord | undefined>(undefined);
+  const [error, setError] = useState<string | undefined>(undefined);
+
+  useEffect(() => {
+    try {
+      let newKey = findKeyById(keys, keyIdOrTag) ?? findKeyByTag(keys, keyIdOrTag)
+      if (newKey === undefined) {
+        throw new Error('Key not found');
+      }
+      setKey(newKey)
+    } catch(error) {
+      logger.error(error);
+      setError((error as Error).message || UNKNOWN_ERROR);
+    }
+  }, [keys, keyIdOrTag])
+
+  const signer = useMemo(() => {
+    try {
+      if (key === undefined) {
+        throw new Error('Key undefined');
+      }
+      const signer = provider.getSigner(key.publicKey)
```

Comment:
If you work with an internally generated wallet `provider.getSigner` will always return you a signer, BUT
if you work with a connected wallet like Metamask you have access to only one account address (key) that is currently selected in the wallet. 
The method you use may lead to an error without checking a currently selected address.

Reply:
So, you have to check the address of the currently selected `account` and if this address is not equal to `publicKey` the signer must be equal to `undefined` (without any errors).